src/polytopes/chain.jl

```diff
@@ -220,56 +191,35 @@ end
 """
     unique(chain)
 
-Return a new `chain` without duplicate vertices.
+Return a new `chain` without duplicate vertices. Closed chains remain closed.
 """
 Base.unique(c::Chain) = unique!(deepcopy(c))
 
-"""
-    close!(chain)
-
-Close the `chain`, i.e. repeat the first vertex
-at the end of the vertex list.
-"""
-function close!(c::Chain)
-  push!(c.vertices, first(c.vertices))
-  c
-end
-
 """
     close(chain)
 
-Same as [`close!`](@ref) but operates on a copy
-of the original `chain`.
+Close the `chain`, i.e. add a segment going from the last to the first vertex.
 """
-Base.close(c::Chain) = close!(deepcopy(c))
-
-"""
-    open!(chain)
-
-Open the `chain`, i.e. remove the last vertex.
-"""
-function open!(c::Chain)
-  pop!(c.vertices)
-  c
-end
+Base.close(c::Chain) = c
+Base.close(c::Ring) = Chain(CircularVector(c.vertices))
 
 """
     open(chain)
 
-Same as [`open!`](@ref) but operates on a copy
-of the original `chain`.
+Open the `chain`, i.e. remove the segment going from the last to the first vertex.
 """
-Base.open(c::Chain) = open!(deepcopy(c))
+# call `open` again to avoid issues in case of nested CircularVector
+Base.open(c::Chain) = open(Chain(parent(c.vertices)))
+Base.open(c::Ring) = c
```

Comment:
Same here, I think the definitions are swapped.

Reply:
Done in newer version.